src/System.Private.CoreLib/shared/System/Convert.cs

```diff
@@ -2492,19 +2494,146 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
+            Vector128<byte> localShuffleMask = s_base64ShuffleMask;
+            Vector128<byte> localTwoBytesStringMaskLo = s_base64TwoBytesStringMaskLo;
+
+            for (; i <= length - stride; i += stride)
+            {
+                // input = [xxxx|DDDC|CCBB|BAAA]
+                Vector128<byte> inputVector = Sse2.LoadVector128(inData + i);
+
+                // bytes from groups A, B and C are needed in separate 32-bit lanes
+                // in = [DDDD|CCCC|BBBB|AAAA]
+                //
+                //      an input triplet has layout
+                //      [????????|ccdddddd|bbbbcccc|aaaaaabb]
+                //        byte 3   byte 2   byte 1   byte 0    -- byte 3 comes from the next triplet
+                //
+                //      shuffling changes the order of bytes: 1, 0, 2, 1
+                //      [bbbbcccc|ccdddddd|aaaaaabb|bbbbcccc]
+                //           ^^^^ ^^^^^^^^ ^^^^^^^^ ^^^^
+                //                  processed bits
+                inputVector = Ssse3.Shuffle(inputVector, localShuffleMask);
+
+                // unpacking
+
+                // t0      = [0000cccc|cc000000|aaaaaa00|00000000]
+                Vector128<byte> t0 = Sse2.And(inputVector, tt0);
+                // t1      = [00000000|00cccccc|00000000|00aaaaaa]
+                Vector128<byte> t1 = Sse2.MultiplyHigh(t0.AsUInt16(), tt1).AsByte();
+                // t2      = [00000000|00dddddd|000000bb|bbbb0000]
+                Vector128<byte> t2 = Sse2.And(inputVector, tt2);
+                // t3      = [00dddddd|00000000|00bbbbbb|00000000]
+                Vector128<byte> t3 = Sse2.MultiplyLow(t2.AsUInt16(), tt3).AsByte();
+                // indices = [00dddddd|00cccccc|00bbbbbb|00aaaaaa] = t1 | t3
+                Vector128<byte> indices = Sse2.Or(t1, t3);
+
+                // lookup function "Single pshufb method" (lookup_pshufb_improved)
+                Vector128<byte> result = Sse2.SubtractSaturate(indices, tt5);
+                Vector128<sbyte> compareResult = Sse2.CompareGreaterThan(tt7, indices.AsSByte());
+                result = Sse2.Or(result, Sse2.And(compareResult.AsByte(), tt8));
+                result = Ssse3.Shuffle(localShiftLut, result);
+                result = Sse2.Add(result, indices);
+                // end of lookup function
+
+                // save as two-bytes string, e.g.:
+                // 1,2,3,4,5..16 => 1,0,2,0,3,0..16,0
+                Sse2.Store(outputBytes + j, Ssse3.Shuffle(result, localTwoBytesStringMaskLo));
+                j += Vector128<byte>.Count;
+
+                // Do it for the second part of the vector (rotate it first in order to re-use asciiToStringMaskLo)
+                result = Sse2.Shuffle(result.AsUInt32(), 0x4E /*_MM_SHUFFLE(1,0,3,2)*/).AsByte();
+                result = Ssse3.Shuffle(result, localTwoBytesStringMaskLo);
+
+                if (insertLineBreaks && (charcount += 16) >= base64LineBreakPosition)
```

Comment:
Having the side effect only hit if `insertLineBreaks` is true, but required for both the true and false scenarios is non-obvious.

It would be nice to move the `charCount += 16` out separately